File: queries/_test_fixtures/typescript/methods.ts
/**
 * TypeScript methods fixture for AST fixture testing.
 */

class Calculator {
    add(a: number, b: number): number {
        return a + b;
    }

    subtract(a: number, b: number): number {
        return a - b;
    }

    multiply(a: number, b: number): number {
        return a * b;
    }

    divide(a: number, b: number): number {
        if (b === 0) {
            throw new Error("Division by zero");
        }
        return a / b;
    }

    calculate(operation: string, a: number, b: number): number {
        const method = (this as any)[operation];
        return method.call(this, a, b);
    }
}

File: queries/_test_fixtures/go/methods.go
package example

// Calculator handles arithmetic.
type Calculator struct {
	Precision int
}

// Add returns a + b.
func (c *Calculator) Add(a int, b int) int {
	return a + b
}

// Subtract returns a - b.
func (c *Calculator) Subtract(a int, b int) int {
	return a - b
}

// Multiply returns a * b.
func (c *Calculator) Multiply(a int, b int) int {
	return a * b
}

// Divide returns a / b.
func (c *Calculator) Divide(a int, b int) (int, error) {
	if b == 0 {
		return 0, fmt.Errorf("division by zero")
	}
	return a / b, nil
}
